jbmc/regression/jbmc-strings/CProverStringToString/Main.java

```diff
@@ -0,0 +1,184 @@
+import org.cprover.CProverString;
+
+class Main {
+  public void testBooleanSuccess(boolean b) {
+    String s = CProverString.toString(b);
+    assert (b && s.equals("true")) || (!b && s.equals("false"));
+  }
+
+  public void testBooleanFailure() {
+    String s = CProverString.toString(true);
+    assert s.equals("false");
+  }
+
+  public void testCharSuccess(int choice) {
+    char c;
+    String r;
+
+    switch(choice) {
+    case 0:
+      c = 'a';
+      r = "a";
+      break;
+    case 1:
+      c = ' ';
+      r = " ";
+      break;
+    case 2:
+      c = '"';
+      r = "\"";
+      break;
+    case 3:
+      c = 'デ';
+      r = "デ";
+      break;
+    case 4:
+      c = Character.MIN_VALUE;
+      r = "\u0000";
+      break;
+    /*case 5:
+      c = Character.MAX_VALUE;
+      r = "\uFFFF";
+      break;*/
+    default:
+      return;
+    }
+
+    String s = CProverString.toString(c);
+    assert s.equals(r);
+  }
+
+  public void testCharFailure() {
+    String s = CProverString.toString('a');
+    assert s.equals("b");
+  }
+
+  public void testIntSuccess(int choice) {
+    int i;
+    String r;
+
+    switch(choice) {
+    case 0:
+      i = 0;
+      r = "0";
+      break;
+    case 1:
+      i = 1;
+      r = "1";
+      break;
+    case 2:
+      i = -1;
+      r = "-1";
+      break;
+    case 3:
+      i = Integer.MIN_VALUE;
+      r = "-2147483648";
+    case 4:
+      i = Integer.MAX_VALUE;
+      r = "2147483647";
+    default:
+      return;
+    }
+
+    String s = CProverString.toString(i);
+    assert s.equals(r);
+  }
+
+  public void testIntFailure() {
+    String s = CProverString.toString(123);
+    assert s.equals("0");
+  }
+
+  public void testLongSuccess(int choice) {
+    long l;
+    String r;
+
+    switch(choice) {
+    case 0:
+      l = 0L;
+      r = "0";
+      break;
+    case 1:
+      l = 1L;
+      r = "1";
+      break;
+    case 2:
+      l = -1L;
+      r = "-1";
+      break;
+    case 3:
+      l = Long.MIN_VALUE;;
+      r = "-9223372036854775808";
+      break;
+    case 4:
+      l = Long.MAX_VALUE;
+      r = "9223372036854775807";
+      break;
+    default:
+      return;
+    }
+
+    String s = CProverString.toString(l);
+    assert s.equals(r);
+  }
+
+  public void testLongFailure() {
+    String s = CProverString.toString(9223372036854775807L);
+    assert s.equals("0");
+  }
+
+  public void testFloatSuccess(int choice) {
+    float f;
+    String r;
+
+    switch(choice) {
+    case 0:
+      f = 0F;
+      r = "0.0";
+      break;
+    case 1:
+      f = 1F;
+      r = "1.0";
+      break;
+    case 2:
+      f = -1F;
+      r = "-1.0";
+      break;
+    case 3:
+      f = 1.1F;
+      r = "1.1";
+      break;
+    /*case 4:
+      f = Float.MAX_VALUE;
+      r = "3.4028235E38";
+      break;
+    case 5:
+      f = Float.MIN_VALUE;
+      r = "1.4E-45";
```

Comment:
Where did this number come from?

Reply:
That's the result of `String.valueOf(f)` (same as `Float.toString(f)`) :bread: